### PR TITLE
Add a Gallery

### PR DESCRIPTION
**Pull request by philippjfr**
@ceball I realize you're probably not done here, but I wanted to add a PR so we can discuss. I'll add some initial examples shortly.

### CODE REVIEW

**Comment by ceball** [comment]
Ha, I typed up the following but when I tried to submit got an error about such a PR already existing. I was very confused until I realized what had happened!

**Comment by ceball** [comment]
This is how I tried to open the PR:

@philippjfr, do you want to add specific, new "gallery" notebooks for the gallery? (That's what I've started to do here, copying holoviews' scheme, and a notebook from holoviews as a placeholder.)

Or should we generate a gallery from the existing notebooks?

Also, can/does any of this relate to 'topics' or 'showcase'? Datashader has topics (https://bokeh.github.io/datashader-docs/topics/index.html), while holoviews has (unused?) showcase.

@jbednar and/or @jlstevens might want to weigh in too (thinking about pyviz projects in general).

There may be an existing issue about this somewhere. Sorry if I missed that.

**Comment by philippjfr** [comment]
>@philippjfr, do you want to add specific, new "gallery" notebooks for the gallery?

Yes, I've written 15 gallery examples so far and will add them shortly.

>Also, can/does any of this relate to 'topics' or 'showcase'?

Currently I don't have any plans for topics or showcase notebooks in GeoViews so we can leave this for now. We should add a topics section to HoloViews soon though.

**Comment by philippjfr** [comment]
@ceball Is the extra nesting in the gallery examples required, i.e. can we get rid of the second ``notebooks`` directory in the path: ``notebooks/gallery/notebooks/matplotlib/area_chart.ipynb``.

**Comment by philippjfr** [comment]
Just added my initial gallery examples, let me know if there's a good reason for the extra nesting and I should move them into the other paths. I'll let you decide and remove the HoloViews examples once then.

**Comment by ceball** [comment]
The extra nesting's just copied from holoviews, where there's gallery/notebooks and gallery/apps. Doesn't seem necessary to me, but maybe I'm missing something?

**Comment by ceball** [comment]
@philippjfr, do you happen to know why I might see this (on macos) while trying to generate gallery thumbnails for some notebooks:

```
/Users/cball/anaconda3/envs/ioamdocenv/lib/python3.6/site-packages/holoviews/plotting/mpl/__init__.py:30: FutureWarning: 'pandas.tseries.converter.register' has been moved and renamed to 'pandas.plotting.register_matplotlib_converters'. 
  converter.register()
Traceback (most recent call last):
  File "<stdin>", line 56, in <module>
NameError: name 'plot_opts' is not defined
brexit_referendum thumbnail export failed
```

If you have no idea, I'll investigate.

**Comment by philippjfr** [comment]
Looks like a bug I didn't catch, will push a fix in a second.

**Comment by ceball** [comment]
(Note that I haven't tried to run the notebooks via the notebook server yet, so it could alternatively be something from the gallery generation.)

**Comment by philippjfr** [comment]
>The extra nesting's just copied from holoviews, where there's gallery/notebooks and gallery/apps. Doesn't seem necessary to me, but maybe I'm missing something?

Just checked and they're broken into apps and demos. That is actually somewhat useful to separate it into sections.

**Comment by philippjfr** [comment]
Know of any nice geographic plots to replicate? I've mostly been working my way through the cartopy gallery. I still have to add attributions and cross-links between bokeh and matplotlib notebooks.

**Comment by ceball** [comment]
> Just checked and they're broken into apps and demos. That is actually somewhat useful to separate it into sections.

Ok. It's configurable now. For now, I'd like to carry on with the structure you used here; we can change it later.

**Comment by ceball** [comment]
Pushes on this PR appear at https://ioam-docs.github.io/geoviews-master/gallery/index.html

**Comment by philippjfr** [comment]
What's the easiest way to add jQuery? It doesn't seem to be loading causing a number of issues.

**Comment by philippjfr** [comment]
Nevermind, works now.

**Comment by jbednar** [comment]
> Know of any nice geographic plots to replicate? 

I don't have anything specific in mind, but anything with about 50 different plots overlaid onto the same map will help reassure people they can make the same incomprehensible map plots I saw everywhere in the AMS meeting this week.  :-)

(Seriously, I think interactive GeoViews+Bokeh plots could really cut down on some of this need to cram *everything* into the same or adjacent plots, by using interactive features to reveal data as needed rather than having "cramming it in" be the only way for data to be available.)

**Comment by philippjfr** [comment]
Thanks for your help @ceball. I'm going to go ahead and merge this for now. I'll probably follow up with tweaks in a new PR.

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ language: python
 
 sudo: false
 
+TODO: update when to upload to ioam-docs before merging
```
[the remark this answers is not in the document]
Comment (philippjfr) [reply .travis.yml]: Bizarre, no idea how that could possibly happen. That said, master seem to have been building okay. Will fix it some time today.

[the remark this answers is not in the document]
Comment (ceball) [reply .travis.yml]: Presumably manual error during merge? Anyway, I like the idea of travis having a TODO key, which you use to tell it what to do in the future. But evidently they haven't implemented that yet...

**Pull request merged**